play-akka-sample-multidc-scala/src/main/scala/sample/persistence/multidc/PaymentLifecycleTracking.scala

```diff
@@ -0,0 +1,78 @@
+package sample.persistence.multidc
+
+import akka.actor.ActorSystem
+import akka.cluster.sharding.{ClusterSharding, ClusterShardingSettings, ShardRegion}
+import akka.persistence.multidc.{PersistenceMultiDcSettings, SpeculativeReplicatedEvent}
+import akka.persistence.multidc.scaladsl.ReplicatedEntity
+import sample.model.PaymentLifecycle.{Command, Event, State}
+
+import scala.reflect.ClassTag
+
+object PaymentLifecycleTracking {
+
+  val shardingName = "PaymentLifecycleTracking"
+  val maxNumberOfShards = 11
+
+  def shardId(entityId: String): String =
+    (math.abs(entityId.hashCode) % maxNumberOfShards).toString
```

Comment:
I think this is wrong when `entityId.hashCode` == `Integer.MIN_VALUE`. 
```suggestion
    (math.abs(entityId.hashCode % maxNumberOfShards)).toString
```

https://github.com/akka/akka/issues/25034